helix-term/src/commands.rs

```diff
@@ -2992,12 +2992,26 @@ pub mod insert {
                         (start, pos, None) // delete!
                     }
                 } else {
-                    // delete char
-                    (
-                        graphemes::nth_prev_grapheme_boundary(text, pos, count),
-                        pos,
-                        None,
-                    )
+                    match (text.get_char(pos.saturating_sub(1)), text.get_char(pos)) {
+                        (Some(_x), Some(_y)) if auto_pairs::DEFAULT_PAIRS.contains(&(_x, _y)) =>
```

Comment:
Use autopairs provided by the document/language rather than the default set:

https://github.com/helix-editor/helix/blob/290b3ebbbe0c365eee436b9de9d6d6fc2b4339e9/helix-term/src/commands.rs#L2816

Reply:
done in https://github.com/helix-editor/helix/pull/2940/commits/e7e591e44095e3e4a898ed1885df04b6dcc2b3c3